00-master.sh

```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/bash
+set -o nounset -o errexit
+
+kubeadm init --pod-network-cidr=10.244.0.0/16 --apiserver-advertise-address=${MASTER_PRIVATE_IP} --apiserver-cert-extra-sans=${MASTER_PUBLIC_IP}
+kubectl --kubeconfig=/etc/kubernetes/admin.conf apply -f https://raw.githubusercontent.com/coreos/flannel/v0.9.1/Documentation/kube-flannel.yml
+systemctl enable docker kubelet
+
+# used to join nodes to the cluster
+kubeadm token create --print-join-command > /tmp/kubeadm_join
```

Comment:
I have seen you mentioned token generation and using `scp` to copy tokens. A more neat way could be to let Terraform generate the token and store it in the variable.

If you're interested, you can potentially use the [`random_string`](https://www.terraform.io/docs/providers/random/r/string.html) feature to generate the token. Token is using the `string1.string2` format, where `string1` is 6 characters and `string2` is 16 characters. Both strings can only take lowercase letters and numbers. Then, you should be able to utilize it in the `deploy.tf` file.

I don't have enough Terraform experience, but something like this should work.